Sources/GEOSwift/GEOS/GEOSHelpers.swift

```diff
@@ -23,19 +23,36 @@ func makePoints(from geometry: GEOSObject) throws -> [Point] {
     guard GEOSCoordSeq_getSize_r(geometry.context.handle, sequence, &count) != 0 else {
         throw GEOSError.libraryError(errorMessages: geometry.context.errors)
     }
+    
+    // Check coordinate dimension to see if we have Z coordinates
+    let dimensions = GEOSGeom_getCoordinateDimension_r(geometry.context.handle, geometry.pointer)
+    let hasZ = dimensions > 2
+    
     return try Array(0..<count).map { (index) -> Point in
         var point = Point(x: 0, y: 0)
         // returns 0 on exception
         guard GEOSCoordSeq_getX_r(geometry.context.handle, sequence, index, &point.x) != 0,
             GEOSCoordSeq_getY_r(geometry.context.handle, sequence, index, &point.y) != 0 else {
                 throw GEOSError.libraryError(errorMessages: geometry.context.errors)
         }
+        
+        if hasZ {
+            var zVal: Double = 0
+            guard GEOSCoordSeq_getZ_r(geometry.context.handle, sequence, index, &zVal) != 0 else {
+                throw GEOSError.libraryError(errorMessages: geometry.context.errors)
+            }
+            point.z = zVal
+        }
+        
         return point
     }
 }
 
 func makeCoordinateSequence(with context: GEOSContext, points: [Point]) throws -> OpaquePointer {
-    guard let sequence = GEOSCoordSeq_create_r(context.handle, UInt32(points.count), 2) else {
+    let hasZ = points.first { $0.z != nil } != nil
```

Comment:
I wish we didn't have to scan the entire array to make this determination. I also don't love that you could have an array with mixed-dimensionality. Would it just be a rule that consumers have to follow that you shouldn't use mixed-dimensionality?